path/libexec.go

```diff
@@ -0,0 +1,19 @@
+package path
+
+import (
+	"fmt"
+	"path/filepath"
+	"runtime"
+)
+
+// PluginBin returns the path of plugin binary file
+func PluginBin(name string) string {
```

Comment:
This does not work with https://github.com/notaryproject/notation-go/blob/main/plugin/manager/manager.go#L79

We need to update plugin.Manager to not need plugin directory if possible. Otherwise, we will need to return both system and user paths for plugins.

Reply:
This part has been redesigned. Please review again.